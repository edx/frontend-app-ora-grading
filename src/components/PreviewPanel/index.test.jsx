import React from 'react';
import { shallow } from 'enzyme';

import { PreviewPanel } from '.';

jest.mock('react-doc-viewer', () => ({
  __esModule: true,
  default: 'DocViewer',
  BMPRenderer: 'BMPRenderer',
  ImageProxyRenderer: 'ImageProxyRenderer',
  JPGRenderer: 'JPGRenderer',
  PDFRenderer: 'PDFRenderer',
  PNGRenderer: 'PNGRenderer',
  TIFFRenderer: 'TIFFRenderer',
  TXTRenderer: 'TXTRenderer',
}));

describe('Preview Panel Component', () => {
  const props = {
    uri: 'some_url.pdf',
  };

  let el;
  beforeEach(() => {
    el = shallow(<PreviewPanel {...props} />);
  });
  test('snapshot', () => {
    expect(el).toMatchSnapshot();
  });

  describe('Component', () => {
    test('Test component render', () => {
      expect(el.isEmptyRender()).toEqual(false);
    });
  });
});
