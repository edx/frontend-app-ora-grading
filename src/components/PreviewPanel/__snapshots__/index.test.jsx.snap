// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Panel Component snapshot 1`] = `
<DocViewer
  className="preview-panel"
  config={
    Object {
      "header": Object {
        "disableFileName": true,
        "disableHeader": true,
      },
    }
  }
  documents={
    Array [
      Object {
        "uri": "some_url.pdf",
      },
    ]
  }
  pluginRenderers={
    Array [
      [MockFunction BMPRenderer],
      [MockFunction ImageProxyRenderer],
      [MockFunction JPGRenderer],
      [MockFunction PDFRenderer],
      [MockFunction PNGRenderer],
      [MockFunction TIFFRenderer],
      [MockFunction TXTRenderer],
    ]
  }
/>
`;
